pkgs/build-support/libredirect/test.c

```diff
@@ -76,6 +82,20 @@ int main(int argc, char *argv[])
     assert(access(TESTPATH, X_OK) == 0);
 
     assert(stat(TESTPATH, &testsb) != -1);
+#ifndef __APPLE__
+    assert(stat64(TESTPATH, &testsb64) != -1);
+#endif
+    assert(fstatat(123, TESTPATH, &testsb, 0) != -1);
+#ifndef __APPLE__
+    assert(fstatat64(123, TESTPATH, &testsb64, 0) != -1);
+#endif
+#ifdef __linux__
+    assert(statx(123, TESTPATH, 0, STATX_ALL, &testsbx) != -1);
+#endif
+
+    assert(getcwd(buf, PATH_MAX) != NULL);
```

Comment:
This made me realize - at some point we're going to need to redirect/rewrite the response of `getcwd` too or some code's going to get very confused.

Reply:
I had that thought too. There's a certain amount of surrealism that libredirect introduces that makes it hard for whatever it wraps to understand where on the filesystem it's operating. Even anything that looks in /proc/mounts with a redirected path could get confused by where it _physically_ exists.

Given the amount of hackery involved, I think my personal inclination -- and how I'm using libredirect -- is that it should really only be used to work around crappy builds that can't be patched normally. But I do see it's also used when running some distributed binaries. It feels like only a matter of time until those do something really unexpected.

Another thought I had was that perhaps it should present the redirected paths as symlinks to the real ones. I know some software is picky about symlinks though.

All of that said, I can add a `getcwd` wrapper to this PR if you think it'd be good to pair with the `chdir` one. But I think policy-wise it's worth reevaluating whether this is something we want to ship outside of build environments too. What do you think?

Reply:
No I agree, we should generally add things as we find the need.